src/HydraAuction/Tx/Common.hs

```diff
@@ -318,7 +320,18 @@ toSlotNo ptime = do
   either (error . show) return $ slotFromUTCTime timeHandle utcTime
 
 autoCreateTx :: AutoCreateParams -> Runner Tx
-autoCreateTx (AutoCreateParams {..}) = do
+autoCreateTx params = do
+  body <- autoCreateTxBody params
+  pure $ signTx (signers params) body
```

Comment:
It should use `authoredUtxos` signers as well.

Reply:
Can you explain how we are using `authoredUtxos`? We use it to fill `TxExtraKeyWitnesses`, but that requires a VKey, we are passing in an SKey and every user will only have access to their own SKey.

I can not find the official docs on `TxExtraKeyWitnesses`, but my understanding is that you specify extra VKeys that will need to sign the tx. For example, the bidder could construct a Tx which spends the collateral of the delegate, and would need to add the delegate VKey as extra key witness? Maybe I am getting it wrong though, could you also clarify what that field means in the tx then?

Reply:
> Can you explain how we are using authoredUtxos?

It is bunch of UTxOs paired with keys to sign them. That API makes clear which UTxO uses which keys and prevents error on missing signing witness or signing.